src/vscode-bicep/src/panes/deploy/app/components/hooks/useAzure.ts

```diff
@@ -216,21 +218,57 @@ function parseError(error: UntypedError) {
 }
 
 function getDeploymentProperties(
+  scope: DeploymentScope,
   metadata: TemplateMetadata,
   paramValues: Record<string, ParamData>,
 ): Deployment {
   const parameters: Record<string, unknown> = {};
-  for (const [key, { value }] of Object.entries(paramValues)) {
-    parameters[key] = {
-      value,
-    };
+  for (const { name } of metadata.parameterDefinitions) {
+    if (paramValues[name]) {
+      const value = paramValues[name].value;
+      parameters[name] = { value };
+    }
   }
 
+  const location =
+    scope.scopeType !== "resourceGroup" ? scope.location : undefined;
+
   return {
+    location,
     properties: {
       mode: "Incremental",
       parameters,
       template: metadata.template,
     },
   };
 }
+
+function getScopeId(scope: DeploymentScope) {
+  switch (scope.scopeType) {
+    case "resourceGroup":
+      return `/subscriptions/${scope.subscriptionId}/resourceGroups/${scope.resourceGroup}`;
+    case "subscription":
+      return `/subscriptions/${scope.subscriptionId}`;
+  }
```

Comment:
Should this throw (or otherwise raise an error) if an unsupported or unexpected scope type is requested?

Reply:
Because I'm using a discriminated object type [here](https://github.com/Azure/bicep/blob/main/src/vscode-bicep/src/panes/deploy/models.ts#L34-L44), TS is actually happy without including a `default` case (normally this would block compilation).